### PR TITLE
Mock repository logic

### PR DESCRIPTION
**Pull request by neelp1030**
Addition of data structures, functions (WIP), hardcoded values to inventory, quotas, and transaction repositories for mock demo purposes. Also refactoring repository related data classes to be within the respective repositories.

### CODE REVIEW

**Review by adshayanB — CHANGES_REQUESTED** [review]
This looks good. Perfect for the demo use case

My question is after this, will the code that connects to firebase and ingest data be written in here or I guess the code that reads the db. And a write will be done somewhere else. 

If so a quick comment to say //Todo: Need to link firebase reads can be useful, just so we know we have to come back and work on this soon (I can work with you on the firebase readings after the demo)

File: app/src/main/java/com/example/farmeraid/data/InventoryRepository.kt [hunk 1]
```diff
@@ -7,14 +7,36 @@ import kotlinx.coroutines.flow.flow
...
-                HomeModel.Produce(
-                produceName = "Test Produce",
-                produceAmount = 5,
-            ))
-            emit(inventoryList)
+            emit(inventory)
         }.flowOn(Dispatchers.IO)
     }
+
+    fun addNewProduce(produceName: String, produceAmount: Int) {
```
Comment (adshayanB) [inline app/src/main/java/com/example/farmeraid/data/InventoryRepository.kt:25]: To do: I think this function should accept a user field as well.

Comment (adshayanB) [reply app/src/main/java/com/example/farmeraid/data/InventoryRepository.kt:25]: Please add a comment to mention this

**Review by adshayanB — APPROVED** [review]
Just small nit, feel free to merge after u add that comment

**Review by SoumilDharaskar — COMMENTED** [review]
Looks good!

**Pull request merged**